hal/src/nRF52840/exflash_hal.c

```diff
@@ -150,6 +150,25 @@ int hal_exflash_init(void)
     return 0;
 }
 
+int hal_exflash_uninit(void)
+{
+    hal_exflash_lock();
+    nrfx_qspi_uninit();
+    // PATCH: Initialize CS pin, external memory discharge 
+    nrf_gpio_cfg_output(QSPI_FLASH_CSN_PIN);
+    nrf_gpio_pin_set(QSPI_FLASH_CSN_PIN);
+    nrf_gpio_cfg_input(QSPI_FLASH_IO1_PIN, NRF_GPIO_PIN_PULLDOWN);
+    // The nrfx_qspi driver doesn't clear pending IRQ
+#if MODULE_FUNCTION != MOD_FUNC_BOOTLOADER
+    SPARK_ASSERT(sd_nvic_ClearPendingIRQ(QSPI_IRQn) == NRF_SUCCESS);
+#else
+    NVIC_ClearPendingIRQ(QSPI_IRQn);
```

Comment:
This conditional should be unnecessary as `sd_nvic_ClearPendingIRQ()` in bootloader will map to `NVIC_ClearPendingIRQ()` directly.